app/controllers/claims/schools/accept_grant_conditions_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Claims::Schools::AcceptGrantConditionsController < Claims::ApplicationController
+  include Claims::BelongsToSchool
+
+  before_action :authorize_grant_conditions
+
+  def show; end
+
+  def update
+    Claims::School::AcceptGrantCondition.call(school: @school, user: current_user)
+    redirect_to claims_school_claims_path
+  end
+
+  private
+
+  def authorize_grant_conditions
+    authorize @authorize_grant_conditions || Claims::AcceptGrantConditions
+  end
+end
```

Comment:
Hmm, this controller name doesn't look very RESTful.

```suggestion
class Claims::Schools::GrantConditionsController < Claims::ApplicationController
  include Claims::BelongsToSchool

  before_action :authorize_grant_conditions

  def show; end

  def accept
    Claims::School::AcceptGrantCondition.call(school: @school, user: current_user)

    redirect_to claims_school_claims_path
  end

  private

  def authorize_grant_conditions
    authorize [:grant_conditions, @school] # Will lookup GrantConditions::SchoolPolicy
  end
end
```

As for the policy, we should pass through the `@school` object within a namespace: https://github.com/varvet/pundit?tab=readme-ov-file#policy-namespacing

```ruby
GrantConditions::SchoolPolicy
```

We want to lookup the policy of an action made on the `@school` object by `current_user`. Therefore, the last argument needs to be the resource affected:

```ruby
authorize [:grant_conditions, @school] # @school is important here.
```

We should then ensure that the person accepting the grant _belongs_ to the school.